pkg/api/pod/warnings.go

```diff
@@ -299,6 +301,13 @@ func warningsForPodSpecAndMeta(fieldPath *field.Path, podSpec *api.PodSpec, meta
 			} else {
 				allPorts[k] = []portBlock{{field: fldPath.Child("ports").Index(i), port: port}}
 			}
+			if port.Name != "" {
+				if other, found := allPortsNames[port.Name]; found {
+					warnings = append(warnings, fmt.Sprintf("%s: duplicate port name %q with %s", fldPath.Child("ports").Index(i), port.Name, other))
```

Comment:
the mechanics of detecting the duplicate lgtm, can someone from networking give any suggestions for what to warn the problem with the overlap might be?

Reply:
> can someone from networking give any suggestions for what to warn the problem with the overlap might be?

cc @aojea @danwinship

Reply:
this is the same as https://github.com/kubernetes/kubernetes/pull/122252#discussion_r1434242445 . 

what about `services and probes that select ports by name will use %s` name of the port name here

Reply:
```suggestion
					warnings = append(warnings, fmt.Sprintf("%s: duplicate port name %q with %s, services and probes that select ports by name will use %s", fldPath.Child("ports").Index(i), port.Name, other, other))
```
How about this, If there is only the warning `only the services and probes that select ports by name will use %s`, I think it is not complete enough. It only tells the impact but not the reason. @liggitt @aojea

Reply:
oh, yeah, I meant amending it to existing message , 👍